tests/unit/s2n_x509_validator_custom_oids_test.c

```diff
@@ -0,0 +1,239 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include "api/unstable/custom_x509_extensions.h"
+#include "crypto/s2n_libcrypto.h"
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "tls/s2n_config.h"
+
+#define S2N_SINGLE_OID_CERT_CHAIN "../pems/custom_oids/single_oid_cert_chain.pem"
+#define S2N_SINGLE_OID_KEY        "../pems/custom_oids/single_oid_key.pem"
+
+#define S2N_MULTIPLE_OIDS_CERT_CHAIN "../pems/custom_oids/multiple_oids_cert_chain.pem"
+#define S2N_MULTIPLE_OIDS_KEY        "../pems/custom_oids/multiple_oids_key.pem"
+
+const char *invalid_oid[] = { "1.3.4.2" };
+const char *single_oid[] = { "1.3.187.25240.2" };
+const char *multiple_oids[] = { "1.3.187.25240.2", "1.3.187.25240.3" };
+const uint32_t multiple_oid_count = s2n_array_len(multiple_oids);
+
+int main(int argc, char *argv[])
+{
+    BEGIN_TEST();
+
+    {
+        /* Safety Check */
+        {
+            EXPECT_FAILURE_WITH_ERRNO(s2n_config_add_custom_x509_extension(NULL, single_oid[0], strlen(single_oid[0])),
+                    S2N_ERR_INVALID_ARGUMENT);
+
+            DEFER_CLEANUP(struct s2n_config *test_config = s2n_config_new_minimal(), s2n_config_ptr_free);
+            EXPECT_NOT_NULL(test_config);
+            EXPECT_FAILURE_WITH_ERRNO(s2n_config_add_custom_x509_extension(test_config, NULL, 0), S2N_ERR_INVALID_ARGUMENT);
+        }
+
+        if (!s2n_libcrypto_supports_custom_oid()) {
+            /* Ensure custom extensions can't be added when linked to an unsupported libcrypto. */
+            {
+                DEFER_CLEANUP(struct s2n_config *test_config = s2n_config_new_minimal(), s2n_config_ptr_free);
+                EXPECT_NOT_NULL(test_config);
+                EXPECT_FAILURE_WITH_ERRNO(s2n_config_add_custom_x509_extension(test_config, single_oid[0], strlen(single_oid[0])),
+                        S2N_ERR_API_UNSUPPORTED_BY_LIBCRYPTO);
+            }
+
+            END_TEST();
+        }
+
+        /* Ensure s2n_config_add_custom_x509_extension() handles multiple extensions correctly. */
```

Comment:
This is already tested in your self talk test, so I'm not sure this test is adding much value anymore.